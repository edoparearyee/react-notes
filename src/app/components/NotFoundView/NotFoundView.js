import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './NotFoundView.scss';

class NotFoundView extends Component {
  constructor(route) {
    super();
    this.location = route.location;
  }
  render() {
    return (
      <div className="NotFoundView">
        <h1>Page Not Found</h1>
        <p>
          No match for <code>{this.location.pathname}</code>
        </p>
      </div>
    );
  }
}

NotFoundView.propTypes = {
  location: PropTypes.shape({
    pathname: PropTypes.string.isRequired,
  }).isRequired,
};

export default NotFoundView;
