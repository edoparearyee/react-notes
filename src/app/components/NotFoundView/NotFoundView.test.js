import React from 'react';
import { shallow } from 'enzyme';
import NotFoundView from './NotFoundView';

describe('NotFoundView', () => {
  it('renders component', () => {
    const component = shallow(<NotFoundView location={{ pathname: '/foo' }} />);
    expect(component).toMatchSnapshot();
  });
});
