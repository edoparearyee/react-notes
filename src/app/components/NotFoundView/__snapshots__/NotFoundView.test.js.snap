// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundView renders component 1`] = `
<div
  className="NotFoundView"
>
  <h1>
    Page Not Found
  </h1>
  <p>
    No match for 
    <code>
      /foo
    </code>
  </p>
</div>
`;
